### PR TITLE
update main github readme

### PR DESCRIPTION
**Pull request by changhiskhan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,71 +25,121 @@ Lance makes machine learning workflows with ML data easy (images, videos, point
 
 * Version, compare and diff ML datasets easily.
 
+* Search for nearest neighbors in under 1 millisecond.
```
Comment (eddyxu) [inline README.md:28]: The example is not clear that `per-query-latency` is under 1 milliseconds.

File: README.md [hunk 2]
```diff
@@ -27,71 +25,121 @@ Lance makes machine learning workflows with ML data easy (images, videos, point
...
 
 ## Benchmarks
 
+### Vector search
+
+We used the sift dataset to benchmark our results with 1M vectors of 128D
+
+1. For 100 randomly sampled query vectors, we get <1ms average response time (on a 2023 m2 macbook air)
+
+![img.png](img.png)
```
Comment (eddyxu) [inline README.md:135]: In the image, change the dataset file name to `sift_1m.lance`?

Comment (changhiskhan) [reply README.md:135]: i think in the tarball the directory is vec_data.lance, so that's what ppl will see by default

Comment (eddyxu) [reply README.md]: oh, lemme make a new tarball, this is the SIFT dataset, right

**Review by eddyxu — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -27,71 +25,121 @@ Lance makes machine learning workflows with ML data easy (images, videos, point
...
 Lance is powered by Lance Format, an Apache-Arrow compatible columnar data format which is an alternative to Parquet, Iceberg and Delta. Lance has 50-100x faster query performance for ML data.
 
+
 ## Quick Start
 
+
+**Converting to Lance**
+
+```python
+import lance
```
Comment (jaichopra) [inline README.md:41]: We have the import before installation, require `pip install pylance` above this

File: README.md [hunk 4]
```diff
@@ -27,71 +25,121 @@ Lance makes machine learning workflows with ML data easy (images, videos, point
...
 **Encodings**: to achieve both fast columnar scan and sub-linear point queries, Lance uses custom encodings and layouts.
 
 **Nested fields**: Lance stores each subfield as a separate column to support efficient filters like “find images where detected objects include cats”.
 
-**Versioning / updates** (ROADMAP): a Manifest can be used to record snapshots. Updates are supported via write-ahead logs.
+**Versioning**: a Manifest can be used to record snapshots. Currently we support creating new versions automatically via appends, overwrites, and index creation 
+
+**Fast updates** (ROADMAP): Updates will be supported via write-ahead logs.
+
+**Vector index**: Vector index for similarity search over embedding space
```
Comment (jaichopra) [inline README.md:122]: Perhaps move this to the first one? as the key value prop

File: README.md [hunk 5]
```diff
@@ -27,71 +25,121 @@ Lance makes machine learning workflows with ML data easy (images, videos, point
...
-ds = lance.dataset("s3://eto-public/datasets/oxford_pet/oxford_pet.lance")
-duckdb.query('select class, count(1) from ds group by 1').to_arrow_table()
+tbl = dataset.to_table()  # next release of duckdb will have pushdowns enabled
+duckdb.query("SELECT * FROM tbl LIMIT 10").to_df()
 ```
 
-You can easily import a DataFrame or a Parquet file to Lance using Apache Arrow-first APIs:
+**Vector search**
+
+Download an indexed [sift dataset](https://eto-public.s3.us-west-2.amazonaws.com/datasets/sift/sift_ivf256_pq16.tar.gz),
```
Comment (jaichopra) [inline README.md:78]: Could this be moved to wget/curl + unzip via shell? or you prefer this for general os - windows included?

**Pull request merged**